addons/print/models/print_model.py

```diff
@@ -0,0 +1,295 @@
+# -*- coding: utf-8 -*-
+import base64
+import re
+import time
+
+from openerp import api, fields, models
+from openerp.tools.safe_eval import safe_eval
+from openerp.tools.translate import _
+
+
+class print_provider(models.Model):
+    """ Print Provider Base Model. Each specific provider can extend the model by adding
+        its own fields, using the provider_name (provider field) as a prefix for the new
+        fields and method.
+    """
+
+    _name = 'print.provider'
+    _rec_name = 'name'
+    _description = 'Print Provider'
+
+    def _get_providers(self):
+        return []
+
+    # --------------------------------------------------
```

Comment:
Unnecessary comment, as now every class will have fields / constraints / methods in a good order.